janus_server/src/aggregator.rs

```diff
@@ -1077,6 +1080,18 @@ impl VdafOps {
             ));
         }
 
+        // Reject reports after a task has expired.
+        // https://www.ietf.org/archive/id/draft-ietf-ppm-dap-02.html#section-4.3.2
+        if report.metadata().time().is_after(task.task_expiration())
+            || clock.now().is_after(task.task_expiration())
```

Comment:
I think we should probably only check the client timestamp here, not the server timestamp.

Otherwise, delayed reports might be arbitrarily denied, even if Collection of the given batch hasn't started yet & we have not yet deleted the task.

(N.B. we might choose to eventually delete all data related to the task, at which point neither upload nor collection would expect to succeed -- at that point we'd need some way to represent the task as "deleted", and that's what we'd check here.)